lsp-types/src/Language/LSP/Protocol/Types/Common.hs

```diff
@@ -187,3 +190,24 @@ instance Semigroup s => Semigroup (s |? Null) where
 k .=? v = case v of
   Just v' -> [fromString k J..= v']
   Nothing -> mempty
+
+{- |
+Parse a value optionally, with lenient but careful handling of 'Null':
```

Comment:
Is it possible to make this comment slightly clearer? Something like
```
If 'null' can be converted to the Null type from the JSON package then the result is 'Just JSON.Null'
If 'null' cannot be converted to the Null type from the JSON package then the result is 'Nothing'
```
maybe?

Reply:
okay, I think I made it clearer!